app/src/main/java/org/dhis2/utils/customviews/AlertBottomDialog.kt

```diff
@@ -0,0 +1,126 @@
+package org.dhis2.utils.customviews
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.FrameLayout
+import androidx.core.content.ContextCompat
+import androidx.databinding.DataBindingUtil
+import com.google.android.material.bottomsheet.BottomSheetBehavior
+import com.google.android.material.bottomsheet.BottomSheetDialog
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment
+import org.dhis2.R
+import org.dhis2.databinding.AlertBottomDialogBinding
+
+class AlertBottomDialog : BottomSheetDialogFragment() {
+
+    companion object {
+        val instance: AlertBottomDialog
+            get() = AlertBottomDialog()
+    }
+
+    fun setTitle(title: String): AlertBottomDialog {
+        this.title = title
+        return this
+    }
+
+    fun setMessage(message: String): AlertBottomDialog {
+        this.message = message
+        return this
+    }
+
+    fun setPositiveButton(text: String? = null, onClick: (() -> Unit)? = null): AlertBottomDialog {
+        this.positiveText = text
+        this.positiveOnclick = onClick
+        return this
+    }
+
+    fun setNegativeButton(text: String? = null, onClick: (() -> Unit)? = null): AlertBottomDialog {
+        this.showNegative = true
+        this.negativeText = text
+        this.negativeOnclick = onClick
+        return this
+    }
+
+    fun setEmptyMandatoryFields(emptyMandatoryFields: List<String>): AlertBottomDialog {
+        this.emptyMandatoryFields = emptyMandatoryFields
+        return this
+    }
```

Comment:
All this setXXXX methods can benefit from the _.apply{ }_ high-order function.
```kotlin
fun setMessage(message: String) = apply { this.message = message }

fun setEmptyMandatoryFields(fields: List<String>) = apply { this.emptyMandatoryFields = emptyMandatoryFields }
```

Reply:
Nice, ok :+1: